### ISSUE TITLE
Change event on original textarea not fired

### ISSUE THREAD

**Issue by cblock** [tiff/wysihtml5#189]
Hi!

first of all thanks for this great editor!
I just noticed that no change event is fired on the original textarea while the wysihtml5 change event fires properly. Is this a bug or a feature?

Background: 
- Upon input "change" I need to change some parent elements of the original textarea (marking them with some css classes). 
- Using the wysihtml5 change event for this purpose is not feasible due to the sandboxing feature of the editor (I can't crawl up the dom tree beyond the editor's iframe boundary and I understand that this is good and important). 
- Hooking into the change event of the original textarea to accomplish this task would be a perfectly feasible solution but unfortunately no event is fired

**Comment by pamelafox-coursera** [comment]
Here's what I did to the TextArea setValue method in the wysi code to get a change event to fire:

``` javascript
  setValue: function(html, parse) {
    if (parse) {
      html = this.parent.parse(html);
    }
    if (html != this.element.value) {
        this.element.value = html;
        var evt = document.createEvent("HTMLEvents");
        evt.initEvent('change', true, true);
        this.element.dispatchEvent(evt);
    }
  },
```

**Comment by hejsan** [comment]
I also have to rely on the change event of the original textarea in my backbone application to populate my models.

**Comment by cblock** [comment]
Any chance that Pull Request #242 gets merged any time soon?

**Comment by hejsan** [comment]
Actually I just tried #242 and it fires onchange every second or so, no matter if anything has changed or not.
Not exactly optimal performance..

**Comment by hejsan** [comment]
ok, more often actually since the editor syncs every 400 milliseconds.

A more robust solution would only sync when things actually change, but it might be more complex than it seems.

**Comment by hejsan** [comment]
I was able to figure out a workaround that works for me:
hook into the "change:composer" event

```
editor.on("change:composer", function(e) {
      var evt = document.createEvent("HTMLEvents");
      evt.initEvent('change', true, true);
      this.textareaElement.dispatchEvent(evt);
});
```

It's still not a real change event though since it only fires on blur...

**Comment by cblock** [comment]
Hi @hejsan 

Thanks for helping out with the workaround. I will test it out and I will also try to come up with a solution that does not poll for changes.

**Comment by hejsan** [comment]
Hi @cblock 

Great to hear it.

**Comment by cblock** [comment]
@hejsan 
I just noticed that `change:composer` seems to be fired only when blurring the focus with mouse. If I navigate away from the text wysihtml5 widget by tab key. Nothing is fired. Can you confirm that?

**Comment by hejsan** [comment]
Yes, if you look closely you will see I mentioned that in the last line of my comment above: https://github.com/xing/wysihtml5/issues/189#issuecomment-10551395 :)

```
It's still not a real change event though since it only fires on blur...
```

**Comment by cblock** [comment]
Nevermind. The `composer:change` event is fired no matter if an object loses focus via mouse or keyboard. It was an error in my implementation.

But to be clear: My understanding is that `change` and `blur` events are both fired only after an element loses focus, right? The difference is that `blur` is fired every time while `change` is only fired if an actual change of the contents occurred.

**Comment by hejsan** [comment]
Yes, I believe so
